Python/Product/VSInterpreters/CPythonInterpreterFactoryProvider.cs

```diff
@@ -100,7 +118,8 @@ class CPythonInterpreterFactoryProvider : IPythonInterpreterFactoryProvider {
                 // Python key no longer exists, so go back to watching
                 // Software.
                 e.CancelWatcher = true;
-                StartWatching(e.Hive, e.View);
+                _registryTags.Remove(e.Tag);
```

Comment:
Do you need to also do RegistryWatcher.Instance.Remove ?

Reply:
No - cancelling via the event removes it from the watcher.